dune/grid/CpGrid.hpp

```diff
@@ -863,6 +864,52 @@ namespace Dune
             return ret;
         }
 
+        /// \brief Get the cartesian tag associated with a face tag.
+        ///
+        /// The tag tells us in which direction the face would point
+        /// in the underlying cartesian grid.
+        /// \param An iterator that points to the face and was obtained
+        /// by iterating over Opm::UgGridHelpers::cell2Faces(grid).
+        template<class Cell2FacesRowIterator>
+        int
+        faceTag(const Cell2FacesRowIterator& cell_face) const
+        {
+            const int cell = cell_face.getCellIndex();
+            const int face = *cell_face;
+            assert (0 <= cell);  assert (cell < numCells());
+            assert (0 <= face);  assert (face < numFaces());
+
+            typedef cpgrid::OrientedEntityTable<1,0>::row_type F2C;
+
+            const cpgrid::EntityRep<1> f(face, true);
+            const F2C&     f2c = current_view_data_->face_to_cell_[f];
+            const face_tag tag = current_view_data_->face_tag_[f];
+
+            assert ((f2c.size() == 1) || (f2c.size() == 2));
+
+            const bool normal_is_in =
+                ((f2c.size() == 1)
+                 ? ! f2c[0].orientation() // Single cell => boundary
+                 : (f2c[0].index() != cell));     // Two cells => interior
+
+            switch (tag) {
+            case LEFT:
+                //                    LEFT : RIGHT
+                return normal_is_in ? 0    : 1; // min(I) : max(I)
```

Comment:
These statements depend on an implementation detail (that is also used in the boundaryId() method): that all faces are created oriented in the positive IJK directions. The same is done by the code that creates face tags for UnstructuredGrid (cornerpoint_grid.c of opm-core), but there is a comment there about this.

I think this PR is fine otherwise, but I'd like to see a comment about this to avoid future developers misunderstanding things.